releasenotes/notes/0.23/vf2_custom_score_analysis-abb191d56c0c1578.yaml

```
@@ -24,4 +24,32 @@ features:
     error rate for gates that operate on (0, 1) is 0.01, and (1, 0) is not
     supported by the target. This will be used for scoring if it's set for the
     ``vf2_avg_error_map`` key in the property set when :class:`~.VF2Layout` and
-    :class:`~.VF2PostLayout` are run.
+    :class:`~.VF2PostLayout` are run. For example::
+
+        from qiskit.transpiler import AnalysisPass, PassManager, Target
+        from qiskit.transpiler.passes import VF2Layout
+        from qiskit.transpiler.passes.layout.vf2_utils import ErrorMap
+        from qiskit.circuit.library import CZGate, UGate
+        from qiskit.circuit import Parameter
+
+        class CustomVF2Scoring(AnalysisPass):
+            error_map = ErrorMap(3)
+            error_map.add_error((0, 0), 0.0024)
+            error_map.add_error((0, 1), 0.01)
+            error_map.add_error((1, 1), 0.0032)
+            return error_map
+
+        target = Target(num_qubits=2)
+        target.add_instruction(
+            UGate(Parameter('theta'), Parameter('phi'), Parameter('lam')),
+            {(0,): None, (1,): None}
+        )
+        target.add_instruction(
+            CZGate(), {(0, 1): None}
+        )
+
+        vf2_pass = VF2Layout(target=target, seed=1234568942)
+        pm = PassManager([FakeScore(), vf2_pass])
```

Some sort of copy-paste bug has gone on here - the class body looks like a function call, and `FakeScore` isn't defined.

---

Yeah it should be `CustomVF2Scoring()` here, I think I copied this from the tests and messed it up:

```suggestion
        from qiskit.transpiler import AnalysisPass, PassManager, Target
        from qiskit.transpiler.passes import VF2Layout
        from qiskit.transpiler.passes.layout.vf2_utils import ErrorMap
        from qiskit.circuit.library import CZGate, UGate
        from qiskit.circuit import Parameter

        class CustomVF2Scoring(AnalysisPass):
            error_map = ErrorMap(3)
            error_map.add_error((0, 0), 0.0024)
            error_map.add_error((0, 1), 0.01)
            error_map.add_error((1, 1), 0.0032)
            return error_map

        target = Target(num_qubits=2)
        target.add_instruction(
            UGate(Parameter('theta'), Parameter('phi'), Parameter('lam')),
            {(0,): None, (1,): None}
        )
        target.add_instruction(
            CZGate(), {(0, 1): None}
        )

        vf2_pass = VF2Layout(target=target, seed=1234568942)
        pm = PassManager([CustomVF2Scoring(), vf2_pass])
```

---

You've still got a problem in your `class` statement there - its body is a function, and it doesn't implement `AnalysisPass.run`.

---

ah I see it now let me push up a quick fix

---

Should be fixed by https://github.com/Qiskit/qiskit-terra/pull/9438/commits/6ad8b3140547a35ce155bb2aad47828c0db67db7